File: 0567-permutation-in-string/0567-permutation-in-string.java
class Solution {
    static boolean anagram(int a[],int b[]){
        for(int i=0;i<26;i++){
            if(a[i]!=b[i]) return false;
        }
        return true;
    }
    public boolean checkInclusion(String s1, String s2) {
        if(s1.length()>s2.length()) return false;
        int a[]=new int[26];
        int b[]=new int[26];
        int j=0;
        while(j<s1.length()){
            a[s1.charAt(j)-'a']++;
            b[s2.charAt(j)-'a']++;
            j++;
        }
        if(anagram(a,b)) return true;
        int i=0;
        while(j<s2.length()){
            b[s2.charAt(j)-'a']++;
            b[s2.charAt(i)-'a']--;
            if(anagram(a,b)) return true;
            i++;
            j++;
        }
        return false;

    }
}

File: 0567-permutation-in-string/0567-permutation-in-string.cpp
class Solution {
public:
    bool checkInclusion(string s1, string s2) {
        if(s1.length()>s2.length()) return false;
        sort(s1.begin(),s1.end());
        int l=s1.length();
        string substr=s2.substr(0,l);
        string newsubstr=substr;
        sort(newsubstr.begin(),newsubstr.end());
        if(newsubstr==s1) return true;
        int j=l;
        while(j<s2.length()){
            substr.erase(substr.begin());
            substr+=s2[j];
            string newsubstr=substr;
            sort(newsubstr.begin(),newsubstr.end());
            if(newsubstr==s1) return true;
            j++;
            
        }
        return false;
    }
};